GameServerLib/GameObjects/GameObject.cs

```diff
@@ -91,33 +94,43 @@ public virtual void OnCollision(IGameObject collider)
         {
         }
 
-        public void Move(float diff)
+        /// <summary>
+        /// Moves the object to its specified waypoints, updating its coordinate.
+        /// </summary>
+        /// <param name="diff">The amount of milliseconds the object is supposed to move</param>
+        public void Move(float diff, bool useTarget = false)
         {
+            // no waypoints remained - clear the Waypoints
+            if (WaypointIndex >= Waypoints.Count)
+            {
+                Waypoints.RemoveAll(v => v != Waypoints[0]);
```

Comment:
Calling RemoveAll every time is probably not doing good to the CPU. Maybe add to the condition ` && Waypoints.Count > 1`.